msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsMessageManagerImpl.java

```diff
@@ -1261,35 +1261,33 @@ public Attachment createAttachment() {
         return attachment;        
     }
 
-    public void saveMessage(Message message) {
-    	saveMessage(message, true);
+    public Message saveOrUpdateMessage(Message message) {
+        return saveOrUpdateMessage(message, true);
     }
 
-    public void saveMessage(Message message, boolean logEvent) {
-    	saveMessage(message, logEvent, toolManager.getCurrentTool().getId(), getCurrentUser(), getContextId());
+    public Message saveOrUpdateMessage(Message message, boolean logEvent) {
+        return saveOrUpdateMessage(message, logEvent, toolManager.getCurrentTool().getId(), getCurrentUser(), getContextId());
     }
-    
-    public void saveMessage(Message message, boolean logEvent, boolean ignoreLockedTopicForum) {
-        saveMessage(message, logEvent, toolManager.getCurrentTool().getId(), getCurrentUser(), getContextId(), ignoreLockedTopicForum);
+
+    public Message saveOrUpdateMessage(Message message, boolean logEvent, boolean ignoreLockedTopicForum) {
+        return saveOrUpdateMessage(message, logEvent, toolManager.getCurrentTool().getId(), getCurrentUser(), getContextId(), ignoreLockedTopicForum);
     }
-    
-    public void saveMessage(Message message, boolean logEvent, String toolId, String userId, String contextId){
-    	saveMessage(message, logEvent, toolId, userId, contextId, false);
+
+    public Message saveOrUpdateMessage(Message message, boolean logEvent, String toolId, String userId, String contextId){
+        return saveOrUpdateMessage(message, logEvent, toolId, userId, contextId, false);
     }
-    
-    public void saveMessage(Message message, boolean logEvent, String toolId, String userId, String contextId, boolean ignoreLockedTopicForum){
+
+    public Message saveOrUpdateMessage(Message message, boolean logEvent, String toolId, String userId, String contextId, boolean ignoreLockedTopicForum){
         boolean isNew = message.getId() == null;
-        
-        if (!ignoreLockedTopicForum && !(message instanceof PrivateMessage)){                  
-          if (isForumOrTopicLocked(message.getTopic().getBaseForum().getId(), message.getTopic().getId())) {
-              log.info("saveMessage executed [messageId: " + (isNew ? "new" : message.getId().toString()) + "] but forum is locked -- save aborted");
-              throw new LockedException("Message could not be saved [messageId: " + (isNew ? "new" : message.getId().toString()) + "]");
-          }
+
+        if (!ignoreLockedTopicForum && !(message instanceof PrivateMessage) && isForumOrTopicLocked(message.getTopic().getBaseForum().getId(), message.getTopic().getId())) {
+            log.info("saveOrUpdateMessage executed [messageId: " + (isNew ? "new" : message.getId().toString()) + "] but forum is locked -- save aborted");
```

Comment:
log message can be improved:
`log.warn("Forum or Topic is locked for [messageId: {}] not saving", (isNew ? "new" : message.getId().toString()));`

Reply:
Right! Done.